### PR TITLE
Set FT for RRTMGP

### PR DESCRIPTION
**Pull request by nefrathenrici**
This PR removes the hardcoded Float64 for RRTMGP. 
Important note: If the zenith angle is approximately 90 degrees, the shortwave flux is not used.

Two test cases have MSE changes, the plots have been checked

### CODE REVIEW

File: src/parameterized_tendencies/radiation/RRTMGPInterface.jl [hunk 1]
```diff
@@ -1211,20 +1210,28 @@ NVTX.@annotate function update_sw_fluxes!(
...
-# TODO: Change these methods back to using face_sw_flux/face_clear_sw_flux
-# instead of face_sw_flux_up - face_sw_flux_dn
-update_net_fluxes!(radiation_mode, model) =
-    parent(model.face_flux) .=
-        parent(model.face_lw_flux) .+ parent(model.face_sw_flux_up) .-
-        parent(model.face_sw_flux_dn)
+function update_net_fluxes!(_, model)
+    model.face_flux .=
+        ifelse.(
+            model.cos_zenith' .≈ 0.0,
```
Comment (szy21) [inline src/parameterized_tendencies/radiation/RRTMGPInterface.jl:1216]: Could we add the tolerance explicitly so we know what tolerance we are using?

**Review by szy21 — APPROVED** [review]
Thanks Nat! Just a minor comment.

**Comment by sriharshakandala** [comment]
I started the GPU pipeline for this branch here: https://buildkite.com/clima/climaatmos-target-gpu-simulations/builds/195

**Comment by sriharshakandala** [comment]
Looks like this buys us about 21% - 25% reduction in walltime for the longwave and shortwave solvers!

**Review by sriharshakandala — APPROVED** [review]

File: src/parameterized_tendencies/radiation/RRTMGPInterface.jl (hunk 1 above)
Comment (nefrathenrici) [reply src/parameterized_tendencies/radiation/RRTMGPInterface.jl]: The kwarg `rtol = ...` seems to cause GPU compilation errors, so I have left it as a comment - see [here](https://buildkite.com/clima/climaatmos-ci/builds/16873#018dd79c-0fa3-461a-8400-ea130d58e917)

Comment (szy21) [reply src/parameterized_tendencies/radiation/RRTMGPInterface.jl]: Could we just do `model.cos_zenith' <= sqrt(eps(FT))` here? Or this wouldn't work on GPU either?

Comment (nefrathenrici) [reply src/parameterized_tendencies/radiation/RRTMGPInterface.jl]: That should work - i will change it!

**Pull request merged**